supplier/supplier.cc

```diff
@@ -42,21 +45,14 @@ class SupplierServiceImpl final : public Supplier::Service {
   // DB maintains a mapping ID -> vector<VendorInfo*>.
   // Stores pointers to VendorInfos, which are stored in an array
   std::unordered_map<uint32_t, vector<VendorInfo*>> vendor_db_;
-  VendorInfo vendor[4];
+  // use C style array to ensure the validity of the address
+  VendorInfo vendor_[10]; 
```

Comment:
I know you are time constrained, so if you don't have enough time you can ignore this. 

All of this complexity can be avoided if vendor_db_ is an std::vector<std::unique_ptr<VendorInfo>>.  Sure some VendorInfo is duplicated, but they are not very large protos so some duplication is ok.